superset-frontend/src/dashboard/components/SaveModal.jsx

```diff
@@ -193,7 +193,11 @@ class SaveModal extends React.PureComponent {
         }
         modalFooter={
           <div>
-            <Button buttonStyle="primary" onClick={this.saveDashboard}>
+            <Button
+              data-test="modal-save-button"
```

Comment:
This is a good example... the attribute seems like a sensible name. `modal-save-dashboard-button` would be even more explicit, but this seems nice and easy to search for and reason about.